app/controllers/CyaNewUkimsNumberController.scala

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import cats.data
+import cats.data.EitherNec
+import connectors.TraderProfileConnector
+import controllers.actions._
+import models.{NormalMode, TraderProfile, ValidationError}
+import navigation.Navigator
+import pages._
+import play.api.i18n.MessagesApi
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import services.AuditService
+import viewmodels.checkAnswers._
+import viewmodels.govuk.summarylist._
+import views.html.CyaNewUkimsNumberView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class CyaNewUkimsNumberController @Inject() (
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  profileAuth: ProfileAuthenticateAction,
+  val controllerComponents: MessagesControllerComponents,
+  view: CyaNewUkimsNumberView,
+  traderProfileConnector: TraderProfileConnector,
+  navigator: Navigator,
+  auditService: AuditService,
+  sessionRepository: SessionRepository
+)(implicit ec: ExecutionContext)
+    extends BaseController {
+
+  private val errorMessage: String = "Unable to update new Ukims number in Trader profile."
+
+  def onPageLoad(): Action[AnyContent] = (identify andThen profileAuth andThen getData andThen requireData) {
+    implicit request =>
+      // TODO validateUkimsNumber validate the UkimsNumberUpdatePage. Needs a new method with NewUkimsNumberPage TGP-2699
+      TraderProfile.validateUkimsNumber(request.userAnswers) match {
+        case Right(_)     =>
+          val list = SummaryListViewModel(
+            // TODO This needs to be changed to NewUkimsNumberPageSummary.row TGP-2699
+            rows = Seq(UkimsNumberSummary.rowUpdate(request.userAnswers)).flatten
+          )
+          Ok(view(list))
+        case Left(errors) =>
+          logErrorsAndContinue(errorMessage, routes.HomePageController.onPageLoad(), errors)
+      }
+  }
+
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    (for {
+      ukimsNumber                <- handleValidateError(TraderProfile.validateUkimsNumber(request.userAnswers))
+      oldTraderProfile           <- traderProfileConnector.getTraderProfile(request.eori)
+      newTraderProfile           <- Future.successful(oldTraderProfile.copy(ukimsNumber = ukimsNumber))
+      _                           = auditService.auditMaintainProfile(oldTraderProfile, newTraderProfile, request.affinityGroup)
+      _                          <- traderProfileConnector.submitTraderProfile(newTraderProfile, request.eori)
```

Comment:
Do we always have to call submit? Is there a user journey where the user can get to this page and submit the previous ukims number? In the other CYAs we only submit if necessary.

Reply:
Here, we need to call submitTraderProfile to set eoriChange to false.